cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -0,0 +1,125 @@
+{{- if eq .NodePool.Profile "worker-karpenter" }}
+---
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: "{{.NodePool.Name}}"
+spec:
+  {{- if index .NodePool.ConfigItems "scaling_priority"}}
+  weight: {{.NodePool.ConfigItems.scaling_priority}}
+  {{- end}}
+  consolidation:
+    {{- if index .NodePool.ConfigItems "consolidation" }}
+    enabled: {{ or .NodePool.ConfigItems.consolidation true }}
+    {{- else }}
+    enabled: true
+    {{- end }}
+  requirements:
+    - key: "node.kubernetes.io/instance-type"
+      operator: In
+      values:
+{{- range $type := .NodePool.InstanceTypes }}
+      - "{{ $type }}"
+{{- end }}
+    - key: karpenter.sh/capacity-type
+      operator: In
+      values:
+{{if .NodePool.IsSpot}}
```

Comment:
I commented this internally, but to have it reflected here as well. I think we really should consider if defining pools as spot or not really makes sense with Karpenter. My proposal would be that ALL pools have both spot and on-demand and then we control via admission (by injecting node selectors on pods) if spot is the default or if on-demand is. That way we will automatically have a setup with spot falling back to on-demand or just only considering on-demand from the pool if the pods should not be allowed to run on spot.